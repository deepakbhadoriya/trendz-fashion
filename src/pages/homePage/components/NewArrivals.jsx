import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Loader from '../../../components/loader/Loader';
import Product from '../../../components/product/Product';
import allProducts from '../../../fakeData/fakeApiAllProducts';
import productAction from '../../../redux/actions/productAction';
import styles from './NewArrivals.module.scss';

const NewArrivals = () => {
  const dispatch = useDispatch();
  const products = allProducts;

  // useEffect(() => {
  //   dispatch(productAction.getProducts());
  // }, []);

  if (products) {
    return (
      <section className={styles.newArrivals}>
        <h2 className={styles.title}>New Arrivals</h2>
        <div className="row">
          {products.map((product) => (
            <Product product={product} key={product.id} />
          ))}
        </div>
      </section>
    );
  } else {
    return (
      <section className={styles.newArrivals}>
        <h2 className={styles.title}>New Arrivals</h2>
        <div className={styles.products}>
          <Loader />
        </div>
      </section>
    );
  }
};

export default NewArrivals;
