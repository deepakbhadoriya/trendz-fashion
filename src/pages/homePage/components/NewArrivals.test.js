import React from 'react';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

import NewArrivals from './NewArrivals';
import TestEnvironmentWrapper from '../../../test/TestEnvironmentWrapper';

const { mount } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });

describe('<NewArrivals/>', () => {
  it('render correctly (snapshot)', () => {
    const tree = mount(
      <TestEnvironmentWrapper>
        <NewArrivals />
      </TestEnvironmentWrapper>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
