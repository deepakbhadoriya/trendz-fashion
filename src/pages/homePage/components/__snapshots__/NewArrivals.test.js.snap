// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewArrivals/> render correctly (snapshot) 1`] = `
<TestEnvironmentWrapper>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <IntlProvider
          defaultFormats={Object {}}
          defaultLocale="en"
          formats={Object {}}
          locale="en"
          messages={
            Object {
              "electronics": "Electronics",
              "fifty_percent_off": "Get up to 50%",
              "help_center": "Help Center",
              "holiday_season": "For the holiday season",
              "jewellery": "Jewellery",
              "login": "Log in",
              "mensClothing": "Men's clothing",
              "referral_message": "Invite a friend and get a discount of 5%",
              "register": "Register",
              "search": "Search for products",
              "thirty_percent_discount_shirts": "Get up to 30% off shirts",
              "womenClothing": "Women's Clothing",
            }
          }
          onError={[Function]}
          textComponent={Symbol(react.fragment)}
        >
          <NewArrivals>
            <section
              className="newArrivals"
            >
              <h2
                className="title"
              >
                New Arrivals
              </h2>
              <div
                className="products"
              >
                <Loader>
                  <center>
                    Loader
                  </center>
                </Loader>
              </div>
            </section>
          </NewArrivals>
        </IntlProvider>
      </Provider>
    </Router>
  </BrowserRouter>
</TestEnvironmentWrapper>
`;
